ethreceipts/ethreceipts.go

```diff
@@ -821,19 +826,19 @@ func collectOk[T any](in []T, oks []bool, okCond bool) []T {
 // 	return txnLogs
 // }
 
-func groupLogsByTransaction(logs []types.Log, numTxns int) [][]*types.Log {
-	if numTxns == 0 {
-		return [][]*types.Log{}
-	}
-	out := make([][]*types.Log, blockLogsCount(numTxns, logs))
+func groupLogsByTransaction(logs []types.Log) map[string][]*types.Log {
+	var out = make(map[string][]*types.Log)
 	for _, log := range logs {
 		log := log
-		out[log.TxIndex] = append(out[log.TxIndex], &log)
```

Comment:
btw, I do remember writing this function long ago -- and my original implementation was similar to what you have now, but then changed it to place in the TxnIndex ... I can't remember exactly why, but it might have been that some nodes gave these to us in the wrong order or something..? I can't recall, but there was a reason. We can just look out for stuff in future around here if anything with receipt listener is acting up